R/04-normalize.R

```diff
@@ -78,79 +117,73 @@ gimap_normalize <- function(.data = NULL,
 
   message("Normalizing Log Fold Change")
 
-  # Based on log fold change calculations and other handling will go based on the code in:
-  # https://github.com/FredHutch/GI_mapping/blob/main/workflow/scripts/03-filter_and_calculate_LFC.Rmd
-
-  if (gimap_dataset$filtered_data$filter_step_run) {
-    dataset <- gimap_dataset$filtered_data$transformed_log2_cpm
-    pg_ids <- gimap_dataset$filtered_data$metadata_pg_ids$id
-  } else {
-    dataset <- gimap_dataset$transformed_data$log2_cpm
-    pg_ids <- gimap_dataset$metadata$pg_ids
-  }
-
   # Doing some reshaping to get one handy data frame
   lfc_df <- dataset %>%
     as.data.frame() %>%
     dplyr::mutate(pg_ids = pg_ids) %>%
     tidyr::pivot_longer(-pg_ids, values_to = "log2_cpm") %>%
     # Adding on metadata
-    dplyr::left_join(gimap_dataset$metadata$sample_metadata, by = c("name" = "col_names")) %>%
-    dplyr::select(-name) %>%
+    dplyr::left_join(gimap_dataset$metadata$sample_metadata %>%
+                       dplyr::select(col_names, comparison),
+                     by = c("name" = "col_names")) %>%
     tidyr::pivot_wider(values_from = "log2_cpm",
-                       names_from = c(timepoints, replicates))
-
+                       names_from = c(name, comparison))
 
+  ##### Checking for missing ids
   missing_ids <- data.frame(
     missing_ids = setdiff(lfc_df$pg_ids, gimap_dataset$annotation$pgRNA_id)
   )
 
   if ((nrow(missing_ids) > 0) & (nrow(missing_ids) < num_ids_wo_annot)){
     message("The following ", nrow(missing_ids), " IDs were not found in the annotation data: \n", paste0(missing_ids, collapse = ", "))
-  } else {
+
+    if (rm_ids_wo_annot){
+      lfc_df <- lfc_df %>%
+        filter(!pg_ids %in% missing_ids$missing_ids)
```

Comment:
The missing_ids weren't fully being dropped because it was a vector inside a data.frame so that is fixed here.